src/app/youtube/directives/border.directive.ts

```diff
@@ -0,0 +1,19 @@
+import { Directive, OnInit, ElementRef, Renderer2, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appBorder]'
+})
+export class BorderDirective implements OnInit {
+  @Input() publishedAt = '';
+
+  constructor(private element: ElementRef, private renderer: Renderer2) { }
+  ngOnInit() {
+    let diff = Math.floor((new Date().getTime() - new Date(this.publishedAt).getTime()) / 86400000);
+    if (diff < 7) { this.renderer.setStyle(this.element.nativeElement.getElementsByClassName('mat-card')[0], 'border-bottom', '3px solid blue'); }
```

Comment:
Please extract all this magic numbers to constants.

Reply:
I extracted